sdk/include/opentelemetry/sdk/trace/batch_span_processor.h

```diff
@@ -134,22 +141,41 @@ class BatchSpanProcessor : public SpanProcessor
   void DrainQueue();
 
 #ifdef ENABLE_ASYNC_EXPORT
-  /* In case of async export, wait and notify for shutdown to be completed.*/
-  void WaitForShutdownCompletion();
+  struct AsyncExportData
+  {
+    nostd::span<std::unique_ptr<Recordable>> recordables;
```

Comment:
Why do we need a span of recordable here? It is just a reference and may get invalidated.